src/util/http.py

```diff
@@ -0,0 +1,136 @@
+from __future__ import annotations
+
+import asyncio
+import typing as t
+from urllib.parse import quote as _uriquote
+
+import aiohttp
+
+from src import errors, log
+
+logger = log.get_logger(__name__)
+
+
+SUCCESS_STATUS: int = 200
+
+
+async def json_or_text(
+    response: aiohttp.ClientResponse,
+) -> dict[str, t.Any] | list[dict[str, t.Any]] | str:
+    """
+    Process an `aiohttp.ClientResponse` to return either a JSON object or raw text.
+
+    This function attempts to parse the response as JSON. If the content type of the response is not
+    application/json or parsing fails, it falls back to returning the raw text of the response.
+
+    Parameters
+    ----------
+    response : aiohttp.ClientResponse
+        The response object to process.
+
+    Returns
+    -------
+    dict[str, t.Any] | list[dict[str, t.Any]] | str
+        The parsed JSON object as a dictionary or list of dictionaries, or the raw response text.
+    """
+    try:
+        if "application/json" in response.headers["content-type"].lower():
+            return await response.json()
+    except KeyError:
+        # Thanks Cloudflare
+        pass
+
+    return await response.text(encoding="utf-8")
+
+
+class Route:
+    """Handle route construction for HTTP requests."""
+
+    def __init__(self, method: str, url: str, **params: int | str | bool) -> None:
+        self.method = method
+
+        if params:
```

Comment:
use yarl for this

Reply:
For what part?

Reply:
Everything related to manipulating URLs, like you do in the highlighted method

Reply:
So for the parameters? I don't get what you are saying as I never used yarl

Reply:
Yes, you are concatenating query parameters by hand, and if i was doing it i'd instead use a trusted library for that

Reply:
Should be fixed.